components/home.tsx

```diff
@@ -1,65 +1,87 @@
 import { useRouter } from "next/router";
 import React, { SyntheticEvent, useCallback, useState } from "react";
-import { GameId, Player } from "./game/types";
-import { PLAYER1, PLAYER2 } from "./game/constants";
+import { GameId, PlayerName } from "./game/types";
 import { useInitGame } from "./game/api/useInitGame";
 
 const useRedirectToGame = () => {
   const router = useRouter();
   return useCallback(
-    (gameId: GameId, player: Player) => {
-      router.push(`/game/${gameId}/${player}`);
+    (
+      gameId: GameId,
+      playerOneName?: PlayerName,
+      playerTwoName?: PlayerName
+    ) => {
+      if (playerOneName !== undefined) {
+        router.push(`/game/${gameId}/${playerOneName}`);
+      } else if (playerTwoName !== undefined) {
+        const res = fetch("/api/game/setPlayerTwo", {
```

Comment:
note1: would player2 be able to join again?

note2: would this mean that player2 name can be changed but player1 name cannot?

Reply:
yes, it is what you say, However the python implementation has the same behavior, so I would like to know @MiltonLn's opinion, should  I refactor it or take another approach?

Reply:
@kevincardonag just try to replicate the behavior from the Python side, we want them to be exactly the same